### DISCUSSION TITLE
.Rprofile in admiral gets changed by styler

### DISCUSSION THREAD

**Discussion by esimms999-gsk** [pharmaverse/admiral#2002] [General]
In admiral@devel, the .Rprofile file in admiral gets changed by styler::style_pkg() and then shows up in the git pane. This is due to some blanks. I think that everything in admiral@devel should go through styler without changes; all pass except .Rprofile.

I think that admiral's .Rprofile comes from admiralci, but I have not figured out how. I expect a fix would need to be done somewhere in that process.

![1_rprofile_01](https://github.com/pharmaverse/admiral/assets/85294667/77942934-e9fe-4d9f-ab7e-09a919095c50)
![1_rprofile_02](https://github.com/pharmaverse/admiral/assets/85294667/7ab2a888-6d5a-482b-b0f7-58ab9337470b)

**Comment by bms63** [comment]
Can we just style it and commit it??

**Comment by esimms999-gsk** [comment]
@bms63  I don't think it is that straight forward. If I understand correctly, the .Rprofile file in admiral comes from admiralci ... and it is not a direct copy. If I "fix" .Rprofile in admiral, it will soon be over-written by some process in admiralci.

**Comment by manciniedoardo** [comment]
@esimms999-gsk is this still a problem?

**Comment by esimms999-gsk** [comment]
@manciniedoardo   I just created a project with admiral, switched to the devel branch and ran styler::style_pkg(). Only .Rprofile gets changed. The lines changed by styler are a bit different now:

![Screenshot 2023-09-21 102040](https://github.com/pharmaverse/admiral/assets/85294667/0b614f1f-11e5-4205-9c44-345357375815)

Interesting that it wants to comment out that source() statement; anyone know what is up with that? I know that source() is frowned upon in packages, but would not expect styler to make such a bold move.

This is not a huge problem ... but it is a bit messy in that styler being run on our production package should be clean, i.e. no changes. I think of it in the same way as if we ran our 697 tests and one of them failed and we said "yeah, we know about that, it isn't important".

**Reply by bms63** [reply]
My understanding is that the `.Rprofile` doesn't ship with the package, just like `renv` and `.git` stuff doesn't ship with package either.  It is purely for us as developers to configure and track our environments.  I believe the impact on users is zero.

If we want to track and attempt to fix, then I suggest we make an issue in `admiralci` and we can attempt to address it.

**Reply by galachad** [reply]
@esimms999-gsk should be fixed in last propagation.

**Comment by bms63** [comment]
Closing as fixed in latest propagation
